pkg/clusteragent/autoscaling/workload/controller_horizontal.go

```diff
@@ -106,22 +100,32 @@ func (hr *horizontalController) performScaling(ctx context.Context, podAutoscale
 		scaleDirection = scaleDown
```

Comment:
Should we handle the case where `targetDesiredReplicas == currentDesiredReplicas` here ?